exonum/src/runtime/dispatcher/schema.rs

```diff
@@ -149,20 +149,42 @@ impl Schema<&Fork> {
             return Err(Error::ServiceIdExists);
         }
 
-        let id = spec.id;
-        let name = spec.name.clone();
+        let instance_id = spec.id;
+        let instance_name = spec.name.clone();
+        let pending_status = InstanceStatus::Active;
+
         instances.put(
-            &name,
+            &instance_name,
             InstanceState {
-                spec: spec.clone(),
-                status: InstanceStatus::Pending,
+                spec,
+                status: None,
+                pending_status: Some(pending_status),
             },
         );
-        instance_ids.put(&id, name);
-        self.pending_instances().push(spec);
+        self.pending_instances().put(&instance_name, pending_status);
+        instance_ids.put(&instance_id, instance_name);
         Ok(())
     }
 
+    /// Adds information about stopping service instance to the schema.
+    pub(crate) fn initiate_stopping_service(&mut self, instance_id: InstanceId) {
```

Comment:
To mirror @popzxc's concerns, it looks sort of inconsistent that `initiate_adding_service` method above returns a `Result` while this one panics. Further, the 2 panics in this method are different: the first one can be triggered by the incorrect input, while the second one is something that should never-ever occur.